uamqp/aio/_client_async.py

```diff
@@ -176,6 +176,23 @@ async def _do_retryable_operation_async(self, operation, *args, **kwargs):
                     absolute_timeout -= (end_time - start_time)
         raise retry_settings['history'][-1]
 
+    async def _keep_alive_worker_async(self):
+        interval = 10 if self._keep_alive is True else self._keep_alive
+        start_time = time.time()
+        try:
+            while self._connection and not self._shutdown:
+                current_time = time.time()
+                elapsed_time = (current_time - start_time)
+                if elapsed_time >= interval:
+                    _logger.info("Keeping %r connection alive. %r",
+                                 self.__class__.__name__,
+                                 self._connection._container_id)
+                    await asyncio.shield(self._connection._get_remote_timeout(current_time))
```

Comment:
I don't think we need to shield this any more?
Now that we're no longer calling into C code - I think simply propagating an Cancelled error is probably fine?
Or was there another reason we were shielding it?

Reply:
nah.. I can't think of reasons to shield it, I copy-pasted from uamqp code.
removed it.